modules/navier_stokes/src/bcs/INSADDisplaceBoundaryBC.C

```diff
@@ -0,0 +1,43 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "INSADDisplaceBoundaryBC.h"
+#include "SystemBase.h"
+#include "ImplicitEuler.h"
+
+registerMooseObject("NavierStokesApp", INSADDisplaceBoundaryBC);
+
+InputParameters
+INSADDisplaceBoundaryBC::validParams()
+{
+  InputParameters params = ADNodalBC::validParams();
+  params.addClassDescription("Boundary condition for displacing a boundary");
+  params.addRequiredCoupledVar("velocity", "The velocity at which to displace");
+  params.addRequiredParam<unsigned short>(
+      "component", "What component of velocity/displacement this object is acting on.");
+  return params;
+}
+
+INSADDisplaceBoundaryBC::INSADDisplaceBoundaryBC(const InputParameters & parameters)
+  : ADNodalBC(parameters),
+    _velocity(adCoupledNodalValue<RealVectorValue>("velocity")),
+    _u_old(_var.nodalValueOld()),
+    _component(getParam<unsigned short>("component"))
+{
+  if (!dynamic_cast<ImplicitEuler *>(_sys.getTimeIntegrator()))
```

Comment:
btw one can write RK schemes as combinations of Euler schemes so technically we would allow those as well if we are careful.
I think this is fine for now